### DISCUSSION TITLE
How do I modify my function so that it return an dataarray with the same  data structure

### DISCUSSION THREAD

**Discussion by delahoussou** [pydata/xarray#8513] [Q&A]

**Comment by keewis** [comment]
to really be able to help you, we'd need a bit more information about the data in the original datasets (`repr`s with explanations would help). However, here are a few comments:
- `.resample` returns a object that allows to apply a reduction method to it, for example `.mean`: `prec.resample(Time="3H").mean()`
- `len` on a `Dataset` / `DataArray` probably doesn't do what you think it does. Instead, compare `.sizes["Time"]` of each object: `resample_prec.sizes["Time"] == resample_temp.sizes["time"] == qrain.sizes["Time"]`
- what does `qrain` contain? If it's integer indices for timesteps, or the values of the timesteps themselves, you could use `resample_temp.[i]sel(Time=qrain)` to build up a condition array that you could pass to `resample_prec.where(..., drop=True)`

**Reply by delahoussou** [reply]
prec_files: represent simulation data for amount of rainfall for every hours for the simulation period, the unit is in "mm" 
```
xarray.Dataset {
dimensions:
	Time = 113952 ;
	south_north = 1015 ;
	west_east = 1359 ;

variables:
	float32 PREC_ACC_NC(Time, south_north, west_east) ;
		PREC_ACC_NC:FieldType = 104 ;
		PREC_ACC_NC:MemoryOrder = XY  ;
		PREC_ACC_NC:description = ACCUMULATED GRID SCALE  PRECIPITATION OVER prec_acc_dt PERIODS OF TIME ;
		PREC_ACC_NC:units = mm ;
		PREC_ACC_NC:stagger =  ;
		PREC_ACC_NC:long_name = PREC_ACC_NC ;
	datetime64[ns] Time(Time) ;
		Time:long_name = Time ;
		Time:description = Time ;
	|S19 Times(Time) ;
		Times:long_name = Time ;
		Times:description = Model output time in yyyy-mm-dd_HH:MM:SS
